coding_systems/dmd/tests/test_import_data.py

```diff
@@ -13,129 +15,151 @@
 from .conftest import MOCK_DMD_IMPORT_DATA_PATH
 
 
-@pytest.fixture(autouse=True)
-def coding_systems_database_tmp_dir(coding_systems_tmp_path):
-    yield coding_systems_tmp_path
-
-
-def test_import_data(
-    settings, dmd_data, dmd_version_asthma_medication, mock_data_download
-):
-    cs_release_count = CodingSystemRelease.objects.count()
-    assert dmd_version_asthma_medication.cached_csv_data == {}
-
-    # import mock XML data
-    # This consists of the AMP 222311000001102 (Ventolin 100micrograms/dose Evohaler)
-    # and its related objects, including:
-    # VMP 39113611000001102 (Salbutamol 100micrograms/dose inhaler CFC free)
-    #  - this VMP has a VMPPREV id 320139002
-    # VPI (with FK to VMP)
-    # AMPP 1479411000001101
-    # VMPP 1056811000001104
-    expected_ids = {
-        AMP: "222311000001102",
-        VMP: "39113611000001102",
-        AMPP: "1479411000001101",
-        VMPP: "1056811000001104",
-    }
-
-    assert not VmpPrevMapping.objects.exists()
-
-    import_data(
-        str(MOCK_DMD_IMPORT_DATA_PATH),
-        release_name="2022 1.0.0",
-        valid_from=date(2022, 10, 1),
-    )
+class TestImportData(DynamicDatabaseTestCase):
+    db_alias = "dmd_2022-100_20221001"
+    import_data_path = str(MOCK_DMD_IMPORT_DATA_PATH)
 
-    # A new CodingSystemRelease has been created
-    assert CodingSystemRelease.objects.count() == cs_release_count + 1
-    cs_release = CodingSystemRelease.objects.latest("id")
-    assert cs_release.coding_system == "dmd"
-    assert cs_release.release_name == "2022 1.0.0"
-    assert cs_release.valid_from == date(2022, 10, 1)
-    # no import ref, defaults to zipfile name
-    assert cs_release.import_ref == "nhsbsa_dmd_1.0.0_20221001000001.zip"
-
-    assert cs_release.database_alias in settings.DATABASES
-
-    for model, expected_id in expected_ids.items():
-        assert model.objects.using("dmd_2022-100_20221001").count() == 1
-        assert model.objects.using("dmd_2022-100_20221001").first().pk == expected_id
-    # VPI has been imported with a FK to the VMP
-    assert VPI.objects.using("dmd_2022-100_20221001").count() == 1
-    assert (
-        VPI.objects.using("dmd_2022-100_20221001").first().vmp
-        == VMP.objects.using("dmd_2022-100_20221001").first()
-    )
+    @pytest.fixture
+    def _get_dmd_version_asthma_medication(self, dmd_version_asthma_medication):
+        self.dmd_version_asthma_medication = dmd_version_asthma_medication
```

Comment:
thought: As mentioned in some of the commit comments, this is a pattern that's needed to get access to a fixture value from within the class. If there's a better way, it's fine to switch to that.